docs/product/roadmap.rst

```diff
@@ -1,6 +1,237 @@
+
+Multiple Types Support
+----------------------
+Users should be able to perform these actions in both the GUI and API:
+
+- Change the type of data per-column to any type supported by Postgres or PostGIS
+
+  - The data in the column will be validated.
+- Import data via any supported import method and have the types of data automatically detected during the import process
+
+  - Types are suggestions and can be changed by the user.
+
+Boolean Type
+------------
+- Add new type (use Postgres type if possible):
+
+  - Boolean
+- Autodetect this type during import
+- Allow user to change columns to this type
+- Add additional grouping options by value (yes/no)
+
+New Text Types
+--------------
+- Add new types:
+
+  - Email
+  - URL
+- Autodetect these types during import
+- Allow user to change columns to these types
+- Add additional grouping options:
+
+  - Email: Domain
+  - URL: TLD, Protocol
+
+Numeric Types
+-------------
+- Add new types, using existing Postgres types where possible:
+
+  - Money (with specified currency)
+  - Percentage
+  - Number
+- Autodetect these types during import
+- Allow user to change columns to these types
+- Add additional grouping options:
+
+  - Number: Range (calculate range options dynamically based on data, e.g. if data varies from 1-100, ranges could be 1-10, 10-20, etc.)
+
+  - Money: Range, Currency
+  - Percentage: Range
+
+Date and Time Types
+-------------------
+- Add new types, using existing Postgres types where possible:
+
+  - Date & Time
+  - Date
+  - Time
+  - Duration
+- Autodetect these types during import
+- Allow user to change columns to these types
+- Add additional grouping options:
+
+  - Date & Time, Date, Time support all grouping options supported by Postgres EXTRACT function.
+  - Duration: Range
+- Allow filtering using natural language for dates (e.g. "next month")
+
+Location Type
+-------------
+- Add new type, using existing PostGIS type where possible:
+
+  - Location
+- Autodetect this type during import
+- Allow user to change columns to this type
+- Add additional grouping options:
+
+  - Street Address
+  - Country
+  - Administrative Area Level 1 *(in the US, these are states)*
+  - Administrative Area Level 2 *(in the US, these are counties)*
+  - Administrative Area Level 3
+  - Administrative Area Level 4
+  - Administrative Area Level 5 
+  - Locality *(city/town)*
+  - Sublocality *(subdivision of city/town)*
+  - Neighborhood
+  - Postal Code
+  - Latitude
+  - Longitude
+
+The attributes of the location column type are based on results returned by the `Google Maps Geocoding API <https://developers.google.com/maps/documentation/geocoding/overview>`_, since they've done the work of putting addresses into a global format.
+
+Phone Number Type
+-----------------
+- Add new type
+ 
+  - Phone Number
+- Autodetect this type during import
+- Allow user to change columns to this type
+- Add grouping options:
+
+  - Country Code
+  - Area Code
```

Comment:
We'll want to figure out how to generalize "area code" appropriately for different countries; we'd probably want some third-party package for that since it's quite different in different contexts (for example, German numbers don't all have the same number of digits in different parts).